pkg/transaction/controller.go

```diff
@@ -128,11 +130,21 @@ func (C *Controller) sendSignedTx() {
 	C.transactionForRPC.receiptHash = &r
 }
 
-func (C *Controller) setIntrinsicGas(gasLimit uint64) {
+func (C *Controller) setIntrinsicGas(gasLimit, inputData string) {
```

Comment:
Feels like we're breaking abstraction here by having `ExecuteTransaction` (and by extension this fn) take gasLimit as an option via empty str. Can you handle the optional gasLimit case b4 `ExecuteTransaction` and keep gasLimit to uint64